jpos/src/main/java/org/jpos/q2/Q2.java

```diff
@@ -360,6 +360,84 @@ public static Q2 getQ2(long timeout) {
         return (Q2) NameRegistrar.get(JMX_NAME, timeout);
     }
 
+    public Map<String, Object> getXmlConfigMap() throws Exception {
+        Iterator iter = dirMap.entrySet().iterator();
+        Map<String, Object> config = new HashMap<>();
+        while (iter.hasNext()) {
+            Map.Entry entry = (Map.Entry) iter.next();
+            File   f        = (File)   entry.getKey ();
+            QEntry qentry   = (QEntry) entry.getValue ();
+            if (qentry.isQBean() && qentry.isQPersist() && isXml(f)) {
+                String fname = f.getName();
+                fname = fname.substring(0, fname.length() - 4);
+                ObjectName qname = qentry.getObjectName ();
+                Element e = (Element) server.getAttribute (qname, "Persist");
+                config.put(fname, getElementMap(e));
+            }
+        }
+        return config;
+    }
+
+    private Map<String, Object> getElementMap(Element e) {
+        Map<String, Object> map = new HashMap<>();
+        String className = e.getAttributeValue("class");
+        if (className != null) {
+            map.put("class", className);
+        }
+        List<Element> children = e.getChildren();
+        Map<String, Object> props = null;
+        Map<String, Object> attr = null;
+        for(Element child: children){
+            String key = child.getName();
+            if (key == "property") {
```

Comment:
Shouldn't you use `"property".equals(key)` here ?

Reply:
I think using `==` is a Kotlin idiom ;-)

Reply:
> I think using `==` is a Kotlin idiom ;-)

Hmm, so that looks to be correct, in Koltin "The equals method returns the same result as the “==” operator" while in Java it seems it doesn't, I find that Java behavior to be strange but jPOS is really the first Java based project I've worked on so I'm less familiar with Java semantics.

The programming language I have the most experience in is python however where `==` is a standard value equality check which seems to be fairly common across most programming languages in general.

I regularly work with many different programming languages and usually just expect to catch these sort of issues during testing/review since it's difficult to remember all the minor syntax differences when jumping around between projects.

Reply:
Pushed a fix changing this to `"property".equals(key)`.